tasks/create_app_htpasswd.yml

```diff
@@ -25,6 +25,11 @@
     - secret
     - htpasswd
 
+- name: "Check application htpasswd file status"
+  stat:
+    path: "{{ app_htpasswd_file_path }}"
+  register: app_htpasswd_file
+
 # Create a htpasswd encrypted file with random passwords for users. Password
 # files are generated plain text by default. They will be encrypted in the next
 # step. If the htpasswd file (or user password files) already exists it won't be
```

Comment:
If i understand well you're checking the presence of the htpasswd file only, not the presence of each password file. Note that we will need to delete the htpasswd file manually if we want to add a new user/password. I think it's not an issue but maybe remove "(or user password files)" from the comment?

Reply:
Password files don't need to be checked because the lookup password works as expected! Hence I think this comment is still relevant.

Reply:
ok then I'm misunderstanding something: why is it creating a password mismatch that the htpasswd file is being recreated? I understand that it is being created from the passwords in the password files. So it should be deterministic?

Reply:
When re-running this task, the lookup password sends the password file raw (encrypted) content, and thus the `htpasswd` module thinks passwords changed and tries to update the `htpasswd` file. I think it's the most probable scenario.